unitTests/ui/StandaloneSearchInterfaceTest.ts

```diff
@@ -121,9 +121,28 @@ export function StandaloneSearchInterfaceTest() {
           handleRedirect();
           await promise;
 
-          expect(cmp.redirectToSearchPage).toHaveBeenCalledWith(options.searchPageUri);
+          expect(cmp.redirectToSearchPage).toHaveBeenCalledWith(options.searchPageUri, '');
           done();
         });
+
+        it(`where there is an analytics event prior to performing a redirection,
+        when the execution plan does not return a url (but cancels pending search events),
+        the redirection url contains the original analytics event cause`, async done => {
+          const cause = analyticsActionCauseList.omniboxFromLink;
+          cmp.usageAnalytics.logSearchEvent(cause, {});
+
+          const promise = Promise.resolve(null);
+          spyOn(cmp.queryController, 'fetchQueryExecutionPlan').and.returnValue(promise);
+
+          handleRedirect();
+          cmp.usageAnalytics.cancelAllPendingEvents();
```

Comment:
I tried different approaches to get the [defer code](https://github.com/coveo/search-ui/blob/master/src/ui/Analytics/LiveAnalyticsClient.ts#L213) run inside the test environment, but could not find a way.

I discussed with @lvu285 about writing a functional test, but there are limitations with nightwatch when accessing the redirection url before it gets changed, or retrieving the first analytics event sent on page load.

Instead, I added this line to simulate the effect of the defer code. It's not ideal, but the best I could find,